aioupnp/gateway.py

```diff
@@ -196,19 +196,26 @@ def debug_gateway(self) -> Dict:
     async def discover_gateway(cls, lan_address: str, gateway_address: str, timeout: int = 30,
                                igd_args: OrderedDict = None, loop=None, unicast: bool = None):
         if unicast is not None:
-            return await cls._discover_gateway(lan_address, gateway_address, timeout, igd_args, loop)
+            return await cls._discover_gateway(lan_address, gateway_address, timeout, igd_args, loop, unicast)
+
         done, pending = await asyncio.wait([
             cls._discover_gateway(
                 lan_address, gateway_address, timeout, igd_args, loop, unicast=True
             ),
             cls._discover_gateway(
                 lan_address, gateway_address, timeout, igd_args, loop, unicast=False
-            )], return_when=asyncio.tasks.FIRST_COMPLETED
-        )
-        for task in list(pending):
+            )
+        ], return_when=asyncio.tasks.FIRST_COMPLETED)
+
+        for task in pending:
             task.cancel()
-        result = list(done)[0].result()
-        return result
+        for task in done:
+            try:
+                task.exception()
+            except asyncio.CancelledError:
+                pass
```

Comment:
```else:
  return task.result()```

Reply:
This should not be needed. The return below should always be callable